2016/proposals/rai-vivek.md

```diff
@@ -0,0 +1,360 @@
+platform people use, which software dependency worked and which did not, error
+logs)
+* Enhance the installation-test scripts by adding an option for users to collect
+ results and send them to a server.
+
+## Technical Details
+
+The technical details of the project can be organized into following:
+
+* Collectable information   : machine information, report
+* Backend                   : design of REST API, framework, database, schema
+* Frontend                  : import data, HTML reports, visualizations
+* installation-test scripts : update the scripts to collect info and send request
+* Documentation             : use travis-ci automated build system, follow pep8
+* Testing                   : use travis-ci, coveralls for coverage
+
+An additional objective of this section is to keep the design sweet and simple
+(KISS).
+
+### Collectable information
+
+In this section, we discuss the parameters that can be collected by the SwC’s
+installation-test scripts to help us analyze and improve installation setup for
+future workshops.
+
+The goal is to collect anonymized, non-identifiable machine information and
+status report of the installation-test scripts.
+
+#### Machine information
+
+* Platform (or System)
+* Architecture (or Machine)
+* Node (Hostname), Processor
+* OS (or Distribution), Release version, Name
+
+Most of this information can be collected using the `platform` Python module.
+A snippet of useful commands is shown below.
+
+```
+import platform
+platform.uname() # cross platform
+platform.win32_ver() # windows
+platform.mac_ver() # mac
+platform.linux_distribution() # linux
+```
+
+#### Status report
+
+* Software check status (pass/fail) along with relevant messages
+* Version number of the found (installed) softwares
+* Any error that may have occurred in the script itself
+* Process information
+* Timestamp
+* (Optional) User email
+
+
+The current installation-test script implementation already collects the list of
+*successful* and *failed* checks along with version information and error
+message as available.
+
+### Server
+
+The server will collect the diagnostic report submitted by installation-scripts
+when they are run on the user machine.
+
+A simple design of the setup is discussed below:
+
+#### API
+
+The following steps are involved in the process:
+
+* Installation-script sends a `POST` request to the server endpoint (for example, https://installation.software-carpentry.org/report).
+  > Using Flask web-development framework for Python. A robust, flexible
+  > choice that is easy to develop and maintain. Scalability is not an issue for
+  > our use case.
+
+* Server verifies the data, open database connection, updates the record.
+  > As suggested in the ideas sections, SQLite will be a low-barrier, easier to
+  > maintain database, appropriate for our needs.
+
+* Send a status response back to the client. The response can contain feedback,
+  suggestions or perhaps link to some survey.
```

Comment:
> These instructions are often detailed sufficiently and probably re-written by each workshop (if not available with any previously conducted one).

They're actually inherited from `workshop-template` repository.

> I could only find this install-lesson repository which is created just a week ago and has no content.

@gvwilson is https://github.com/swcarpentry/install-lesson supposed to contain per-dependency installation instructions? Like extended version of `workshop-template` instructions? Because if not, we may want to create such pages for this project.